widget-context.php

```diff
@@ -418,16 +418,10 @@ function context_check_url( $check, $settings ) {
 
 	function match_path( $patterns ) {
 
-		global $wp;
-
 		$patterns_safe = array();
 
-		// Get the request URI from WP
-		$url_request = $wp->request;
-
-		// Append the query string
-		if ( ! empty( $_SERVER['QUERY_STRING'] ) )
-			$url_request .= '?' . $_SERVER['QUERY_STRING'];
+		// Get the request URI
+		$url_request = trim( $_SERVER['REQUEST_URI'], '/' );

```

Comment:
I don't know and I don't have any instances of WP installed in a
sub-directory to test it.

On Wed, Sep 14, 2016 at 8:35 AM, Kaspars Dambis <notifications@github.com>
wrote:

> In widget-context.php
> <https://github.com/kasparsd/widget-context-wporg/pull/20#discussion_r78739126>
> :
>
> >  		$patterns_safe = array();
> >
> > -		// Get the request URI from WP
> > -		$url_request = $wp->request;
> > -
> > -		// Append the query string
> > -		if ( ! empty( $_SERVER['QUERY_STRING'] ) )
> > -			$url_request .= '?' . $_SERVER['QUERY_STRING'];
> > +		// Get the request URI
> > +		$url_request = trim( $_SERVER['REQUEST_URI'], '/' );
>
> @lachmanski <https://github.com/lachmanski> I'm not sure. I personally
> think that your approach is better in terms of user experience and
> expectations. We just need to be careful about not breaking existing
> configurations.
>
> Do you know if $wp->request is relative to WP or the root?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kasparsd/widget-context-wporg/pull/20/files/f6017b4807d87496697046cf9151e2eedd5a3b61#r78739126>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AUQmp-LdL4iyoKYXGgfEHEbpNyTYvyuTks5qp-oNgaJpZM4J8Qlc>
> .
>